modules/CNN_constructor.py

```diff
@@ -0,0 +1,128 @@
+        Could be 'relu', 'sigmoid' or 'swish'.
+    
+    Raises
+    ------
+    ValueError
+        If resnet_type have unknown value
+        
+    Notes
+    -----   
+        [1]_Bag of Tricks for Image Classification with Convolutional Neural Networks. Part 4.
+        https://arxiv.org/pdf/1812.01187.pdf
+        
+    Examples
+    --------
+        ResNetLike([3, 4, 6, 3], 1000, False): 
+            create standart ResNet-34 for ImageNet dataset
+        ResNetLike([3, 4, 6, 3], 1000, True): 
+            create standart ResNet-50 for ImageNet dataset
+        
+        ResNetLike([2, 2, 2, 2], 10, False, resnet_type='D'):
+            create ResNetD-18 for Imagenette dataset
+            
+        ResNetLike([3, 4, 23, 3], 10, False, resnet_type='B', activation='swish'):
+            create ResNetB-101 for Imagenette dataset with swish activation function
+
+    """
+    def __init__(self, 
+                 layers, 
+                 num_classes,
+                 bottleneck,
+                 resnet_type='A',
+                 activation='relu'
+                 ):
+        
+        super().__init__()
+        
+        if resnet_type in ['C', 'D']:
+            self.first = nn.Sequential(
+                nn.Conv2d(3, 32, kernel_size=3, stride=2, padding=1),
+                nn.BatchNorm2d(32),
+                nn.Conv2d(32, 32, kernel_size=3, stride=2, padding=1),
+                nn.BatchNorm2d(32),
+                nn.Conv2d(32, 64, kernel_size=3, stride=1, padding=1),
+                nn.BatchNorm2d(64))
+        elif resnet_type in ['A', 'B']:
+            self.first = nn.Sequential(
+                nn.Conv2d(3, 64, kernel_size=7, stride=2, padding=3),
+                nn.BatchNorm2d(64),
+                nn.MaxPool2d(kernel_size=3, stride=2, padding=1))
+        else:
+            raise ValueError('Unknown resnet_type value')
+        
+        self.body = nn.Sequential()
+        if bottleneck:
+            for num, layer in enumerate(layers):
+                for block in range(layer):
+                    if block == 0  and num < len(layers) - 1:
+                        downsample = 1
+                    elif block == 0 and num == len(layers) - 1:
+                        downsample = -1
+                    elif block != 0:  
+                        downsample = 0
+                    self.body.add_module(name='block_%d_%d'%(num+2,block+1), 
+                                         module=CNN_blocks.ResNetBottleneckBlock(
+                                             num+2, 
+                                             downsample=downsample,
+                                             activation=activation,
+                                             block_type=resnet_type))
+        elif not bottleneck:
+            for num, layer in enumerate(layers):
+                for block in range(layer):
+                    if block == 0  and num < len(layers) - 1:
```

Comment:
The shapes of output  tensors are incorrect because of this line - the first res block does not apply a downsample and should output a tensor with shape (56, 56), while your implementation has (26, 26).
Moreover, using three values for `downsample` seems a bit excessive. Consider refactoring this logic